### DISCUSSION TITLE
Highlight code affected by the current line

### DISCUSSION THREAD

**Discussion by brettcannon** [microsoft/pylance-release#2595] [Enhancement]
Simplest is something like https://github.com/folke/twilight.nvim, but with Pylance knowing call chains and such, highlighting other parts of a file where a line has potential influence would be nice to have to understand the ramification of changes.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by judej** [comment]
@brettcannon, @greazer this looks like functionality that is similar to the Gather preview. Would this make more sense to do this in Gather instead of Pylance?

**Reply by brettcannon** [reply]
I know I wouldn't think about looking in Gather for this, but if the functionality is already there then I guess so?

**Reply by gramster** [reply]
We have the opposite information I think to what you want. We know what code affects _the_ current line, not what code is affected _by_ the current line.

**Comment by bschnurr** [comment]
https://github.com/microsoft/pyrx/issues/2757
